StorageUtility/Windows/DeviceHandle.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#include "Device.h"
+#include "DeviceHandle.h"
+#include "StorageUtility.h"
+
+namespace vtStor
+{
+
+cDeviceHandle::cDeviceHandle(std::shared_ptr<cDeviceInterface> Device)
+{
+    String devicePath;
+    GetDevicePath(devicePath, Device);
+    GetStorageDeviceHandle(devicePath, m_DeviceHandle);
+    //TODO: handle errors
+}
+
+cDeviceHandle::~cDeviceHandle()
+{
+    CloseDeviceHandle(m_DeviceHandle);
```

Comment:
Who is the "user" in this case? If it is the application, I wouldn't give it that responsibility or direct access to the handle. We want to prevent bad programming defensively and not have the "user" closing the handle at the wrong time\place, either by accident or intentionally.
________________________________
From: Trevor Chau<mailto:notifications@github.com>
Sent: ‎8/‎13/‎2015 13:12
To: virtium/vtStor<mailto:vtStor@noreply.github.com>
Cc: phandinhlan<mailto:pdl.dev@outlook.com>
Subject: Re: [vtStor] - Supported Device and DeviceHandle with multiple OS (#30)

> +#include "StorageUtility.h"
> +
> +namespace vtStor
> +{
> +
> +cDeviceHandle::cDeviceHandle(std::shared_ptr<cDeviceInterface> Device)
> +{
> +    String devicePath;
> +    GetDevicePath(devicePath, Device);
> +    GetStorageDeviceHandle(devicePath, m_DeviceHandle);
> +    //TODO: handle errors
> +}
> +
> +cDeviceHandle::~cDeviceHandle()
> +{
> +    CloseDeviceHandle(m_DeviceHandle);

Nothing should be done here. It is the user responsibility to close the device handle.

---
Reply to this email directly or view it on GitHub:
https://github.com/virtium/vtStor/pull/30/files#r37019612